include/nbl/builtin/hlsl/ieee754/impl.hlsl

```diff
@@ -0,0 +1,54 @@
+#ifndef _NBL_BUILTIN_HLSL_IEE754_IMPL_HLSL_INCLUDED_
+#define _NBL_BUILTIN_HLSL_IEE754_IMPL_HLSL_INCLUDED_
+
+#include <nbl/builtin/hlsl/type_traits.hlsl>
+#include <nbl/builtin/hlsl/glsl_compat/core.hlsl>
+#include <nbl/builtin/hlsl/bit.hlsl>
+
+namespace nbl
+{
+namespace hlsl
+{
+namespace ieee754
+{
+
+namespace impl
+{
+template <typename T>
+NBL_CONSTEXPR_INLINE_FUNC bool isTypeAllowed()
+{
+	return is_same<T, uint16_t>::value ||
+		is_same<T, uint32_t>::value ||
+		is_same<T, uint64_t>::value ||
+		is_same<T, float16_t>::value ||
+		is_same<T, float32_t>::value ||
+		is_same<T, float64_t>::value;
+}
```

Comment:
this is best done as an alias type and variable
```cpp
template<typename T>
using is_type_allowed = integral_constant<bool,...>;

template<typename T>
NBL_CONSTEXPR bool is_type_allowed_v = is_type_allowed<T>::value;
```

then its constexpr, even in HLSL